xdsl/utils/lexer.py

```diff
@@ -506,6 +506,7 @@ def lex(self) -> Token:
             f"Unexpected character: {current_char}",
         )
 
+    bare_identifier_regex = re.compile(r"[a-zA-Z_][a-zA-Z0-9_$.]*")
     bare_identifier_suffix_regex = re.compile(r"[a-zA-Z0-9_$.]*")
```

Comment:
It seems like a good idea to do this in a separate PR, with explicit tests, probably reusing the suffix also
```suggestion
    SUFFIX = r"[a-zA-Z0-9_$.]*"
    bare_identifier_regex = re.compile(r"[a-zA-Z_]" + SUFFIX)
    bare_identifier_suffix_regex = re.compile(SUFFIX)
```